layers/core_validation.cpp

```diff
@@ -2931,20 +2974,90 @@ bool CoreChecks::ValidatePipelineUnlocked(const PIPELINE_STATE *pPipeline, uint3
                              string_VkFormat(rendering_struct->stencilAttachmentFormat));
         }
 
-        if ((enabled_features.core11.multiview == VK_FALSE) && (rendering_struct->viewMask != 0)) {
-            skip |=
-                LogError(device, "VUID-VkPipelineRenderingCreateInfo-multiview-06066",
-                         "vkCreateGraphicsPipelines() pCreateInfos[%" PRIu32 "]: multiview is not enabled but viewMask is (%u).",
-                         pipelineIndex, rendering_struct->viewMask);
+        if (rp_state_required) {
+            if ((enabled_features.core11.multiview == VK_FALSE) && (rendering_struct->viewMask != 0)) {
+                skip |= LogError(device, "VUID-VkGraphicsPipelineCreateInfo-multiview-06577",
+                                 "vkCreateGraphicsPipelines() pCreateInfos[%" PRIu32
+                                 "]: multiview is not enabled but viewMask is (%u).",
+                                 pipelineIndex, rendering_struct->viewMask);
+            }
+
+            if (MostSignificantBit(rendering_struct->viewMask) >=
+                static_cast<int32_t>(phys_dev_props_core11.maxMultiviewViewCount)) {
+                skip |= LogError(device, "VUID-VkGraphicsPipelineCreateInfo-renderPass-06578",
+                                 "vkCreateGraphicsPipelines() pCreateInfos[%" PRIu32
+                                 "]: Most significant bit in "
+                                 "VkPipelineRenderingCreateInfo->viewMask(%u) must be less maxMultiviewViewCount(%u)",
+                                 pipelineIndex, rendering_struct->viewMask, phys_dev_props_core11.maxMultiviewViewCount);
+            }
         }
+    }
 
-        if (MostSignificantBit(rendering_struct->viewMask) >=
-            static_cast<int32_t>(phys_dev_props_core11.maxMultiviewViewCount)) {
-            skip |= LogError(device, "VUID-VkPipelineRenderingCreateInfo-viewMask-06067",
-                             "vkCreateGraphicsPipelines() pCreateInfos[%" PRIu32
-                             "]: Most significant bit in "
-                             "VkPipelineRenderingCreateInfo->viewMask(%u) must be less maxMultiviewViewCount(%u)",
-                             pipelineIndex, rendering_struct->viewMask, phys_dev_props_core11.maxMultiviewViewCount);
+    // If VK_EXT_graphics_pipeline_library is not enabled, a complete set of state must be defined at this point
+    if (!IsExtEnabled(device_extensions.vk_ext_graphics_pipeline_library)) {
+        size_t missing_state_msg_size = 0;
```

Comment:
Couldn't you do this into a single std::string like:
```
  std::string msg;
  if (!pPipeline->vertex_input_state) {
           msg += "<vertex input>";
  }
  ...
  if (!msg.empty()) {
        LogErrror();
  }
```

Reply:
👍 yeah, that's what should be done here. I wasn't sure how to properly print this and played around with a few different options, but ended up with something that just looks bad.